iml-gui/crate/src/components/tree.rs

```diff
@@ -51,24 +52,28 @@ fn get_targets_by_parent_resource(
     }
 }
 
-fn get_targets_by_pool_id(cache: &Cache, ostpool_id: u32) -> Vec<&Target<TargetConfParam>> {
+fn get_targets_by_pool_id(cache: &ArcCache, ostpool_id: u32) -> Vec<&Target<TargetConfParam>> {
     let target_ids: Vec<_> = cache
         .ost_pool_osts
-        .values()
+        .arc_values()
         .filter(|x| x.ostpool_id == ostpool_id)
         .map(|x| x.managedost_id)
         .collect();
 
-    cache.target.values().filter(|x| target_ids.contains(&x.id)).collect()
+    cache
+        .target
+        .arc_values()
+        .filter(|x| target_ids.contains(&x.id))
+        .collect()
 }
 
 fn get_targets_by_fs_id(
-    xs: &im::HashMap<u32, Target<TargetConfParam>>,
+    xs: &im::HashMap<u32, Arc<Target<TargetConfParam>>>,
     fs_id: u32,
     kind: TargetKind,
 ) -> Vec<&Target<TargetConfParam>> {
-    xs.values()
-        .filter(|x| match kind {
+    xs.arc_values()
+        .filter(|x: &&Target<TargetConfParam>| match kind {
```

Comment:
It is not related to this PR, but to me it looks like `kind` is matched against `TargetKind` many times, instead of just one.

Reply:
Suggestion?

Reply:
https://github.com/whamcloud/integrated-manager-for-lustre/pull/1572 :)